staging/src/k8s.io/api/networking/v1alpha1/well_known_labels.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+const (
+
+	// LabelIPAddressFamily is used to indicate the IP family of a Kubernetes IPAddress.
+	// This label simplify dual-stack client operations allowing to obtain the list of
+	// IP addresses filtered by family.
+	LabelIPAddressFamily = "ipaddress.kubernetes.io/ip-family"
+	// LabelManagedBy is used to indicate the controller or entity that manages
+	// an IPAddress. This label aims to enable different IPAddress
+	// objects to be managed by different controllers or entities within the
+	// same cluster. It is highly recommended to configure this label for all
+	// IPAddress objects.
+	LabelManagedBy = "ipaddress.kubernetes.io/managed-by"
```

Comment:
Why isn't this is the controller / ownerref?

Reply:
NonNamespaces objects can not be referenced by namespaced objects like Services

> In v1.20+, if the garbage collector detects an invalid cross-namespace ownerReference, or a cluster-scoped dependent with an ownerReference referencing a namespaced kind, 

or is another kind of reference?

Reply:
We need to fix owner ref.  Grr....

Reply:
If we can't fix owner reference, or find an equivalent mechanism that doesn't use labels, could the label key be `kubernetes.io/managed-by`? I'd like to just have one notion of things managing things rather than _n_ of them.

Reply:
that is a good point, we should open this thread with apimachinery about owner references

Reply:
@liggitt can you enlighten us if this is namspace -> non-namespace owneref something that can be fixed, is just nobody worked on it, or is something you don't see we can change it ... I'm willing to spend time on it if this can be fixed since will allow to remove a considerable amount of code from the repair loop for clusterIPs

Ref https://github.com/kubernetes/kubernetes/pull/92743

Reply:
both of the following were intentionally out of scope for owner references:
* cluster-scoped things owned by namespaced things (higher permission thing owned by something in a lower permission scope)
* namespaced things owned by things in other namespaces

both of those lines are permission boundaries, and letting arbitrary objects cross those lines was considered confusing (spooky action at a distance) and/or dangerous (especially with aspects of ownerReferences like blockOwnerDeletion)

see the note at https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications